.travis.yml

```diff
@@ -24,11 +24,35 @@ jobs:
       script: npm run test && npm run e2e-test
       node_js: "8"
     - stage:
-      os: linux
+      os: osx
+      if: branch = master
       env:
         # this is just to indicate in the Travis job list that this is generating docs
-        - GENERATE_DOCS=true
-      script: npm run generate-docs
+        - GENERATE_DOCS=true GITHUB_TOKEN=$TRIAGER_BOT_TOKEN
+      before_script:
+        - brew install hub
+        - hub --version
+        - git remote set-url origin https://triager:$TRIAGER_BOT_TOKEN@github.com/appium/appium
+        - git reset --hard origin/master
+        - git checkout -b generate-docs-$TRAVIS_COMMIT
+        - npm install
+        - npm install -g uncommitted
+
+      script:
+        # Generate the docs
+        - npm run generate-docs
+
+        # If no docs were generated, don't do anything
+        - |
+          if [ !uncommitted ]; then
```

Comment:
Should just be able to check `git diff-index --quiet HEAD --` without needing the npm package.